dev/CommonStyles/ToggleSwitch_themeresources.xaml

```diff
@@ -217,64 +219,118 @@
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchFillOff}" />
                                         </ObjectAnimationUsingKeyFrames>
                                         <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SwitchKnobOff" Storyboard.TargetProperty="Fill">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchKnobFillOffPointerOver}" />
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchKnobFillOff}" />
                                         </ObjectAnimationUsingKeyFrames>
                                         <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SwitchKnobOn" Storyboard.TargetProperty="Fill">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchKnobFillOnPointerOver}" />
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchKnobFillOn}" />
                                         </ObjectAnimationUsingKeyFrames>
                                         <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SwitchKnobBounds" Storyboard.TargetProperty="Fill">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchFillOnPointerOver}" />
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchFillOn}" />
                                         </ObjectAnimationUsingKeyFrames>
                                         <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SwitchKnobBounds" Storyboard.TargetProperty="Stroke">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchStrokeOnPointerOver}" />
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchStrokeOn}" />
                                         </ObjectAnimationUsingKeyFrames>
                                         <ObjectAnimationUsingKeyFrames Storyboard.TargetName="SwitchAreaGrid" Storyboard.TargetProperty="Background">
-                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchContainerBackgroundPointerOver}" />
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ToggleSwitchContainerBackground}" />
                                         </ObjectAnimationUsingKeyFrames>
+                                        <DoubleAnimationUsingKeyFrames Storyboard.TargetName="SwitchKnobOn" Storyboard.TargetProperty="Width" EnableDependentAnimation="True" >
+                                            <SplineDoubleKeyFrame KeyTime="0:0:0.25" KeySpline="0.0,0.0 0.0,1.0" Value="12" />
```

Comment:
Wondering if this number can be defined an x:double somewhere else? (Same with 14 and 17 below.) -- in case it's related to other sizes.

Reply:
> Wondering if this number can be defined an x:double somewhere else? (Same with 14 and 17 below.) -- in case it's related to other sizes.

You could, but the ToggleSwitch is a pretty unique control design, and it does not support resizing like other controls, so it would not be changed commonly.

Having said that, if the Thumb's size at rest is changeable with lightweight styling, the PointerOver, Pressed, and Disabled sizes could be added.  But they would still be something like:
``ToggleSwitchKnobDefaultSize``
``ToggleSwitchKnobPointerOverSize``
``ToggleSwitchKnobPressedSize``
``ToggleSwitchKnobDisabledSize``

Reply:
The 12px and 14px sizes are the same as RadioButtons and SliderThumbs. That also poses the question whether that should those be a shared resource?

Reply:
> The 12px and 14px sizes are the same as RadioButtons and SliderThumbs. That also poses the question whether that should those be a shared resource?

In that case, you could have a generalised value with the pixel sizes - but then use Lightweight styles for the controls, which reference those generalised values.

This would allow developers an easy way to modify the controls, if say they wanted to have the Slider Thumb parts be larger.

It is a little more complicated with some animations as you are putting in scale factors, rather than discreet values.